src/routes/kafka-rest-proxy.ts

```diff
@@ -22,61 +22,73 @@ import urlJoin from 'url-join';
 import logger from '../utils/logger';
 import fetch from 'node-fetch';
 import { json } from 'body-parser';
-import express from 'express';
+import express, { Router } from 'express';
+import { EgoClient } from 'services/ego';
+import authenticatedRequestMiddleware, {
+  AuthenticatedRequest,
+} from 'routes/middleware/authenticatedRequestMiddleware';
 import egoTokenUtils from 'utils/egoTokenUtils';
 
-var router = express.Router();
-const apiRoot = KAFKA_REST_PROXY_ROOT;
+const createKafkaRouter = (egoClient: EgoClient): Router => {
+  const router = express.Router();
+  const apiRoot = KAFKA_REST_PROXY_ROOT;
 
-// fetch needs to use the json body parser
-router.use(json());
+  // fetch needs to use the json body parser
+  router.use(json());
 
-// middleware to secure the kafka proxy endpoint
-// it expects a valid jwt
-router.use((req, res, next) => {
-  const jwt = (req.headers.authorization || '').split(' ')[1] || '';
-  if (jwt === '') {
-    return res.status(401).send({
-      message: 'this endpoint needs a valid jwt token',
-    });
-  }
+  router.use(authenticatedRequestMiddleware({ egoClient }));
+
+  router.post('/:topic', async (req: AuthenticatedRequest, res) => {
+    const {
+      auth: { authenticated, serializedScopes, egoJwt },
+      params: { topic },
+    } = req;
 
-  if (!egoTokenUtils.isValidJwt(jwt)) {
-    return res.status(401).send({
-      message: 'expired token',
+    // Require auth
+    if (!authenticated) {
+      res.status(401).json({ error: 'invalid auth token' });
+      return;
+    }
+
+    // Require kafka topic writing permisison
+    const hasPermission = egoTokenUtils.canWriteKafkaTopic({
```

Comment:
This line is the entire purpose of this PR. All the rest is just clean up on how to pass JWT details into the route handlers.